Cargo.toml

```diff
@@ -7,6 +7,7 @@ members = [
     # FIXME: getting "multiple matching crates for `diesel`" because of the shared build directory
     # "diesel_compile_tests",
     "diesel_tests",
+    "diesel_infer_schema",
```

Comment:
diesel_codegen_syntex and diesel_codegen_shared are dead, as stated in #568. It is not possible to use a proc-macro lib as depenency without using the proc-macro version. Therefore the infer schema related code needs to live somewhere else. A new crate is in my opinion a clearer solution, as putting it into diesel itself.